ctapipe/instrument/camera/readout.py

```diff
@@ -0,0 +1,151 @@
+"""
+import logging
+
+import numpy as np
+from astropy import units as u
+from astropy.coordinates import Angle, SkyCoord
+from astropy.table import Table
+from astropy.utils import lazyproperty
+from scipy.spatial import cKDTree as KDTree
+from scipy.sparse import lil_matrix, csr_matrix
+
+from ctapipe.utils import get_table_dataset, find_all_matching_datasets
+from ctapipe.coordinates import CameraFrame
+
+
+__all__ = ['CameraReadout']
+
+logger = logging.getLogger(__name__)
+
+
+class CameraReadout:
+    """
+    Stores properties related to the readout of a Cherenkov Camera
+
+    Parameters
+    ----------
+    self: type
+        description
+    cam_id: camera id name or number
+        camera identification string
+    sampling_rate
+    reference_pulse_shape
+    reference_pulse_step
+    """
+
+    def __init__(self, cam_id, sampling_rate, reference_pulse_shape, reference_pulse_step):
+        self.cam_id = cam_id
+        self.sampling_rate = sampling_rate
+        self.reference_pulse_shape = reference_pulse_shape
+        self.reference_pulse_step = reference_pulse_step
+
+    def __eq__(self, other):
+        if self.cam_id != other.cam_id:
+            return False
+
+        if self.sampling_rate != other.sampling_rate:
+            return False
+
+        if (self.reference_pulse_shape != other.reference_pulse_shape).all():
+            return False
+
+        if self.reference_pulse_step != other.reference_pulse_step:
+            return False
+
+        return True
+
+    def __hash__(self):
+        return hash((
+            self.cam_id,
+            self.sampling_rate.to_value(u.GHz),
+            self.reference_pulse_shape.size,
+            self.reference_pulse_step.to_value(u.ns),
+        ))
+
+    def __len__(self):
+        return self.reference_pulse_shape.size
+
+    @classmethod
+    def from_name(cls, camera_id='NectarCam', version=None):
+        """
+        Construct a CameraReadout using the name of the camera and array.
+
+        This expects that there is a resource in the `ctapipe_resources` module
+        called "[array]-[camera].camreadout.fits.gz" or "[array]-[camera]-[
+        version].camgeom.fits.gz"
+
+        Parameters
+        ----------
+        camera_id: str
+           name of camera (e.g. 'NectarCam', 'LSTCam', 'GCT', 'SST-1M')
+        version:
+           camera version id (currently unused)
+
+        Returns
+        -------
+        new CameraReadout
+        """
+
+        if version is None:
+            verstr = ''
+        else:
+            verstr = f"-{version:03d}"
+
+        tabname = "{camera_id}{verstr}.camreadout".format(camera_id=camera_id,
+                                                       verstr=verstr)
+        table = get_table_dataset(tabname, role='dl0.tel.svc.camera')
+        return CameraReadout.from_table(table)
+
+    def to_table(self):
+        """ convert this to an `astropy.table.Table` """
+        return Table([self.reference_pulse_shape],
```

Comment:
I don't think using interpolation in integration_correction is correct, it isn't representative of how waveform sampling works, and should result in a different value. 

Switching to a time axis instead of the step is just increasing your data volume unnecessarily, and doesn't simplify the integration_correction calculation.

Reply:
I have renamed `reference_pulse_step` to `reference_pulse_sample_width`

Reply:
> I don't think using interpolation in integration_correction is correct, it isn't representative of how > waveform sampling works, and should result in a different value.

But the reference pulse is already sampled at a higher precision, so you are in effect already doing something like a linear spline, so I wouldn't think it would matter much. But I don't really mind either way, it's just nice to have both columns in the table, so it is unambiguous what sample number each value corresponds to.

Reply:
Imagine the table gets sorted or shuffled, then you would destroy the pulse if you only have the Y value and no X value.

Reply:
Also, for debugging, we should be able to plot the reference pulse without doing any math or using ctapipe at all, just by 

```py3
tab = Table.read("pulse.fits")
with quantity_support():
    plt.scatter(tab['time'], tab['value'])
```
and see the output in nanoseconds on the x-axis

Reply:
Okay, the time axis is now included inside the table